docs/querying/sql-data-types.md

```diff
@@ -71,8 +71,8 @@ Casts between two SQL types with the same Druid runtime type have no effect othe
 Casts between two SQL types that have different Druid runtime types generate a runtime cast in Druid.
 
 If a value cannot be cast to the target type, as in `CAST('foo' AS BIGINT)`, Druid either substitutes a default
-value (when `druid.generic.useDefaultValueForNull = true`, the default mode), or substitutes [NULL](#null-values) (when
```

Comment:
Swap these so the default (SQL-compatible) behavior is first, and explicitly mention that `true` is a legacy mode.